src/main/java/thaumicboots/api/ItemVoidBoots.java

```diff
@@ -0,0 +1,142 @@
+package thaumicboots.api;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.EnumRarity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.DamageSource;
+import net.minecraft.world.World;
+import net.minecraftforge.common.ISpecialArmor;
+
+import baubles.common.lib.PlayerHandler;
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+import taintedmagic.common.registry.ItemRegistry;
+import thaumcraft.api.IWarpingGear;
+import thaumcraft.client.fx.ParticleEngine;
+import thaumcraft.client.fx.particles.FXWispEG;
+import thaumicboots.main.utils.TabThaumicBoots;
+
+public class ItemVoidBoots extends ItemBoots implements IWarpingGear, ISpecialArmor {
+
+    public ItemVoidBoots(ArmorMaterial par2EnumArmorMaterial, int par3, int par4) {
+        super(par2EnumArmorMaterial, par3, par4);
+        setCreativeTab(TabThaumicBoots.tabThaumicBoots);
+        setUnlocalizedName(unlocalisedName);
+    }
+
+    protected void setBootsData() {
+        visDiscount = 5;
+        jumpBonus = 0.450D;
+        tier = 3;
+        iconResPath = "thaumicboots:voidComet_16x";
+        armorResPath = "thaumicboots:model/VoidwalkerBootsComet_-_Purple.png";
+        unlocalisedName = "ItemVoidComet";
+        rarity = EnumRarity.epic;
+    }
+
+    @Override
+    public EnumRarity getRarity(final ItemStack stack) {
```

Comment:
Useless override

Reply:
Nah that one isn't

Reply:
It is, you're already setting the `rarity` field in `setBootsData()`, which you should return in the super method